### PR TITLE
 use eslint and prettier to apply consistent, modern formatting

### PR DESCRIPTION
**Pull request by jstewmon**
I noticed that the style of the source was inconsistent, and that more modern features (block-scope and template literals) were not being leveraged, so I added prettier and eslint, which were able to automatically fix all issues except a couple uses of `var`, which I fixed by hand.

### CODE REVIEW

**Comment by stash** [comment]
@jstewmon thanks for all these pull requests. I'll do a proper review in the near future.

**Comment by jstewmon** [comment]
@stash thanks for the review! I've incorporated all of your feedback.

I used eslint to auto fix function to arrow function where it was able.

I also hand edited the areas you mentioned in your review to improve readability.

**Comment by jstewmon** [comment]
@stash I'm sure you're busy - just dropping a comment in case this fell off your radar.

**Comment by awaterma** [comment]
@jstewmon -- I am currently reviewing this offline. Hopefully should get some time over the weekend or next week to close this out -- or offer some simple changes. Looks very good so far; sorry to take so much time. Once we have this merged, I will move on to the next P/R in this vein. Please see our projects page to track our focus as we ramp up on this project again.

**Comment by jstewmon** [comment]
Thanks, @awaterma! I’m happy to address any feedback you have.

File: package.json [hunk 1]
```diff
@@ -58,17 +58,23 @@
   ],
   "scripts": {
     "version": "genversion lib/version.js && git add lib/version.js",
-    "test": "vows test/*_test.js",
-    "cover": "nyc --reporter=lcov --reporter=html vows test/*_test.js"
+    "test": "npm run eslint && vows test/*_test.js",
```
[the remark this answers is not in the document]
Comment (jstewmon) [reply package.json]: done!

**Comment by awaterma** [comment]
I'm going to merge this on Monday. I've hand reviewed the code, I do see that the changes that @stash requested have been incorporated. I also didn't see any changes to the meaning of the code, just syntax updates. Our tests pass. I think we are good. I'll leave this comment up for a few days in case @stash has some feedback I've missed, but we should be ready to merge and move on to the next p/r.

**Comment by awaterma** [comment]
Most of the differences are in test; will update this thread in a bit with the results of our discussion.

**Comment by jstewmon** [comment]
> In addition to any patterns in a .eslintignore file, ESLint always ignores files in /node_modules/* and /bower_components/*.

https://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories

Any differences you see should be the result of hand edits that I made in response to feedback from @stash which you can review in the discussion on this issue.

I suppose it's also possible that eslint changed some fix formats (or is able to fix lines that it previously could not) since this PR was originally opened.

**Comment by awaterma** [comment]
Thanks for getting back to me @jstewmon! Very sorry for the glacial pace; @stash 's comments about checking the output made me try even harder to look for security issues. We have discussed the diffs, and I think that's all ready to go.

I'm seeing a timing difference due to the coverage directory; I can add something for that later if you're not inclined. Let me know, so I can add an issue to track.

**Comment by jstewmon** [comment]
@awaterma how much time are you seeing on which scripts and what do you expect? I'm seeing `npm run eslint` take 5-6s.

**Comment by awaterma** [comment]
If you run coverage, then run `npm run eslint` on your branch you'll see what I mean. I was seeing execution times a little over 5 minutes. Of course, I am running this in a container. Anyway, I'll just open up an issue and add a .eslintignore file for coverage later. Not a stress; grateful to you for the work. :)

**Comment by awaterma** [comment]
I've added #166 for us to add the ignore file. I'm going to approve and merge this.

**Comment by awaterma** [comment]
Looks like we have failures in the build now; all was good with Travis on the p/r. 

I'll have to sort this out.

@ruoho 

Previous report: https://travis-ci.org/salesforce/tough-cookie/builds/561472603?utm_source=github_status&utm_medium=notification

**Comment by awaterma** [comment]
We should be able to sort this out tomorrow; thanks for finding the old date issue @jstewmon.

**Comment by jstewmon** [comment]
No problem - wasn't sure how you wanted to handle the git mechanics, but I pushed two new commits here - one just fixes the broken tests (looks like there are others that might want similar treatment - exp of 2027); the other prunes an unused variable I spotted in the test file while fixing.

Let me know if there's anything else I can help with. 😄

Once this settles on master, I'll rebase the other PRs I have, so they'll have a concise diff for review.

**Pull request merged**